packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
@jemc I changed the original issue. It's not an attempt to match string add. I'm going to talk to @sylvanc about changing string add to be byteseq rather than string box.